include/aos/common/array.hpp

```diff
@@ -0,0 +1,368 @@
+     * @param buffer underlying buffer.
+     * @param size current array size.
+     */
+    explicit Array(const Buffer& buffer, size_t size = 0)
+        : mSize(size)
+    {
+        SetBuffer(buffer);
+    }
+
+    /**
+     * Creates array instance from const C array.
+     *
+     * @param items const C array.
+     */
+    explicit Array(const T* items)
+        : mItems(items)
+        , mSize(cMaxSize)
+        , mMaxSize(cMaxSize)
+    {
+    }
+
+    /**
+     * Creates array instance from another array.
+     *
+     * @param array another array instance.
+     */
+    Array(const Array& array)
+        : mItems(array.mItems)
+        , mSize(array.mSize)
+        , mMaxSize(array.mMaxSize)
+    {
+    }
+
+    /**
+     * Assigns existing array to the current one.
+     *
+     * @param array existing array.
+     * @return Array&.
+     */
+    Array& operator=(const Array& array)
+    {
+        assert(mMaxSize <= array.mMaxSize);
+
+        mSize = array.mSize;
+
+        if (mItems == array.mItems) {
+            return *this;
+        }
+
+        memcpy(mItems, array.mItems, array.mSize * sizeof(T));
+
+        return *this;
+    }
+
+    /**
+     * Clears array.
+     */
+    void Clear() { mSize = 0; }
+
+    /**
+     * Checks if array is empty.
+     *
+     * @return bool.
+     */
+    bool IsEmpty() const { return mSize == 0; }
+
+    /**
+     * Checks if arrays is full.
+     *
+     * @return bool.
+     */
+    bool IsFull() const { return mSize == mMaxSize; }
+
+    /**
+     * Returns current array size.
+     *
+     * @return size_t.
+     */
+    size_t Size() const { return mSize; }
+
+    /**
+     * Returns maximum available array size.
+     *
+     * @return size_t.
+     */
+    size_t MaxSize() const { return mMaxSize; }
+
+    /**
+     * Sets new array size.
+     *
+     * @param size new size.
+     * @return Error.
+     */
+    Error Resize(size_t size)
+    {
+        if (size > mMaxSize) {
+            return ErrorEnum::eNoMemory;
+        }
+
+        mSize = size;
+
+        return ErrorEnum::eNone;
+    }
+
+    /**
+     * Returns pointer to array items.
+     *
+     * @return T*.
+     */
+    T* Get(void) { return mItems; }
+
+    /**
+     * Returns pointer to const array items.
+     *
+     * @return const T*.
+     */
+    const T* Get(void) const { return mItems; }
+
+    /**
+     * Provides access to array item by index.
+     *
+     * @param index item index.
+     * @return T&.
+     */
+    T& operator[](size_t index) { return mItems[index]; }
+
+    /**
+     * Provides access to array const item by index.
+     *
+     * @param index item index.
+     * @return const T&.
+     */
+    const T& operator[](size_t index) const { return mItems[index]; }
+
+    /**
+     * Provides access to array item by index with boundaries check.
+     *
+     * @param index item index.
+     * @param item return item.
+     * @return Error.
+     */
+    Error At(size_t index, T** item)
```

Comment:
what if return `Pair<Error, T*>` ?
I mean it will be helpful because it's not necessary to create variables and pass them to function